spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/mapping/CommonMappingRuleInterpreter.scala

```diff
@@ -82,60 +108,81 @@ trait CommonMappingRuleInterpreter {
-    * Default values may be specified for each target attribute in a mapping table and must have the same type as
-    * the target attribute and must be presented as a Spark expression string.
-    *
-    * When a mapping table definition has a default value for the target attribute "*", this value acts as a default
-    * value for all target attributes, for which the default value is not set.
-    *
-    * A target attribute used is specified in a mapping rule definition in the list of conformance rules in the dataset.
-    *
-    * @param mappingTableDef A mapping rule definition
-    * @return A default value, if available, as a Spark expression represented as a string.
-    */
-  private def getDefaultValue(mappingTableDef: MappingTable)
-                             (implicit spark: SparkSession, dao: MenasDAO): Option[String] = {
+   * Returns a map of the default values of the output columns, if specified, for a particular mapping rule.
+   * Default values may be specified for each target attribute in a mapping table and must have the same type as
+   * the target attribute and must be presented as a Spark expression string.
+   *
+   * When a mapping table definition has a default value for the target attribute "*", this value acts as a default
+   * value for all target attributes, for which the default value is not set.
+   *
+   * A target attribute used is specified in a mapping rule definition in the list of conformance rules in the dataset.
+   *
+   * @param mappingTableDef A mapping rule definition
+   * @return A map of defaults wit the values as a String Spark expression.
+   */
+  private def getDefaultValues(mappingTableDef: MappingTable)
+                              (implicit spark: SparkSession, dao: MenasDAO): Map[String, String] = {
     val defaultMappingValueMap = mappingTableDef.getDefaultMappingValues
 
-    val attributeDefaultValueOpt = defaultMappingValueMap.get(rule.targetAttribute)
+    val mappingTableSchemaOpt = Option(dao.getSchema(mappingTableDef.schemaName, mappingTableDef.schemaVersion))
     val genericDefaultValueOpt = defaultMappingValueMap.get("*")
+    val defaultValuesForTargets = rule.allOutputColumns().flatMap{case (_, targetAttribute) => {
+      val maybeString = defaultMappingValueMap.get(targetAttribute)
+      maybeString match {
+        case None => genericDefaultValueOpt.map(genDefault => targetAttribute -> genDefault)
+        case Some(x) => Some(targetAttribute -> x)
+      }
+    }}
 
-    val defaultValueOpt = attributeDefaultValueOpt match {
-      case Some(_) => attributeDefaultValueOpt
-      case None => genericDefaultValueOpt
-    }
-
-    if (defaultValueOpt.isDefined) {
-      val mappingTableSchemaOpt = Option(dao.getSchema(mappingTableDef.schemaName, mappingTableDef.schemaVersion))
-      mappingTableSchemaOpt match {
-        case Some(schema) =>
+    mappingTableSchemaOpt match {
```

Comment:
To preserve the original logic here should be a check for non-empty `defaultValuesForTargets`